src/Autonomous.cpp

```diff
@@ -1,227 +1,308 @@
+}
+
+void filterLandmarks(points_t &landmarks)
+{
+	landmarks.erase(std::remove_if(landmarks.begin(), landmarks.end(),
+								   [](point_t p) { return p[2] == 0; }),
+					landmarks.end());
 }
 
 void Autonomous::autonomyIter()
 {
-  if (!Globals::AUTONOMOUS) return;
-  transform_t gps = readGPS(); // <--- has some heading information
-  transform_t odom = readOdom();
-  points_t lidar = readLidarScan();
-  points_t landmarks = readLandmarks();
-
-
-  pose_t gpsPose = toPose(gps, currHeading);
-  currHeading = gpsPose(2);
-
-  // TODO incredibly clever algorithms for state estimation
-  // and path planning and control!
-  //
-  // Ben: better localization than just GPS
-  //    EKF that outputs transforms into a map frame
-  //  input:
-  //    GPS reading, odom reading, old EKF estimate
-  //  output:
-  //    robot frame -> map frame transform
-  //    GPS frame -> map frame transform
-  //
-  // Jonah: map compositing
-  //    taking data from different points in time and combining
-  //    into a single map
-  //  input:
-  //    current lidar scan, past composite list of lidar scans
-  //    robot frame -> map frame transform
-  //  output:
-  //    map (provided as list of points)
-  //    list of locations in map frame for the AR tags we've seen so far
-  //
-  // Atharva: "mission planning" -- higher level planning
-  //    driving through gates
-  //    deciding when we're close enough to posts
-  //  input: internal state, current URCLeg information
-  //    currently visible AR tags in robot frame
-  //    robot frame -> map frame transform
-  //    GPS frame -> map frame transform
-  //  output: goal pose in map frame
-  //
-  // Assaf: taking a map and computing a plan
-  //  input: pose of robot in map frame, map (provided as a list of points), goal pose in map frame
-  //  output: movement command
-  //
-
-  if (arrived(gpsPose)) { //toPose(gps, currHeading)
-    std::cout << "arrived at gate" << std::endl;
-    std::cout << "x: " << gpsPose(0) << " y: " << gpsPose(1) << " theta: " << gpsPose(2) << std::endl;
-
-
-    // implement moving through gates
-    // then update target
-  } else {
-    double dtheta = pathDirection(lidar, gpsPose);
-    double speed = 10.0;
-    // TODO incredibly clever algorithms for state estimation
-    // and path planning and control!
-    
-    if(!lidar.empty()) {
-        speed = 5.0;
-    }
-    if (!Globals::E_STOP) setCmdVel(dtheta, speed);
-  }
-}
-
-double Autonomous::pathDirection(points_t lidar, pose_t gpsPose) {
-    double dtheta;
-    if(lidar.empty()) {
-        dtheta = angleToTarget(gpsPose);
-    } else {
-        const std::vector<PointXY>& ref = points_tToPointXYs(lidar);
-        PointXY direction = pather.getPath(ref, (float) gpsPose(0), (float) gpsPose(1), target);
-        float theta = atan2(direction.y, direction.x);
-        dtheta = static_cast<double>(theta) - gpsPose(2);
-    }
-    return dtheta;
+	if (!Globals::AUTONOMOUS)
+		return;
```

Comment:
If we're going to put this on a separate line, I'd suggest adding `{}` around it. Otherwise it's easy to mistakenly add extra indented lines here without realizing that only the first one is conditioned.